### DISCUSSION TITLE
Windowed mode doesn't work in C#

### DISCUSSION THREAD

**Discussion by laheller** [rokups/rbfx#728] [Q&A]
Hi @slapin @cadaver @wibbe @nemerle @friesencr 

I have the following rbfx C# code:
```csharp
    partial class MyRBFXApp : Application {
        public MyRBFXApp(Context context) : base(context) {
            SysConsole.WriteLine("Hello world!");
        }

        [STAThread]
        static void Main(string[] args) {
            Launcher.Run(_ => new MyRBFXApp(_));
        }

        protected override void Dispose(bool disposing) {
            Context.Renderer.SetViewport(0, null);
            base.Dispose(disposing);
        }

        public override void Setup() {
            EngineParameters[Urho3D.EpRenderBackend] = (int)RenderBackend.OpenGl;
            EngineParameters[Urho3D.EpFullScreen] = false;
            EngineParameters[Urho3D.EpWindowWidth] = 800;
            EngineParameters[Urho3D.EpWindowHeight] = 600;
            EngineParameters[Urho3D.EpWindowResizable] = false;
            EngineParameters[Urho3D.EpWindowTitle] = "Ladsi";
            EngineParameters[Urho3D.EpApplicationName] = "LadsiApp";
            EngineParameters[Urho3D.EpOrganizationName] = "hu.laheller";
            //base.Setup();
        }

        public override void Start() {
			// do something here
            base.Start();
        }

        public override void Stop() {
			// cleanup
            base.Stop();
        }
    }

```
Problem is that no matter what I configure in **EngineParameters**, the application always starts in fullscreen, even when the
_EngineParameters[Urho3D.EpFullScreen] = false;_

Tried different combinations, backends, etc. but it ends up always in fullscreen.
Where is the problem?

BR,

Ladislav

**Comment by gleblebedev** [comment]
That is an interesting list of mentions you have. I'm curious about what will happen next.

What nuget package do you use? What version of the package?

**Reply by laheller** [reply]
Hi @gleblebedev 

I am using exactly [this](https://www.nuget.org/packages/unofficial.Urho3DNet/0.3.7.845/) nuget package version which seems to be yours.

BR,

Ladislav

**Comment by gleblebedev** [comment]
Yes, looks like a bug on my side. I'll look into it, thank you for reporting.

**Reply by laheller** [reply]
@gleblebedev 

Please let me know here, once fixed and a new nuget package is released.
Thanks.

Oh, by the way, what exactly the below highlighted log line menas:

> [20:52:43] [debug] [main] : Created directory C:/Users/hellerl1/AppData/Roaming/hu.laheller/LadsiApp/
> **[20:52:43] [info] [main] : Resource root file is not found or invalid**
> [20:52:43] [info] [main] : ConfigFile 'EngineParameters.json' is not found
> [20:52:43] [info] [main] : ConfigFile overrides 'conf://EngineParameters.json' are loaded
> 

It's an INFO log, but anyway, what Resource Root File it tries to access?
Just for my information :-)

**Reply by laheller** [reply]
@gleblebedev 

OK, nevermind, the above I already figured out from the source code, even if I am not a C++ developer :-)
https://github.com/rbfx/rbfx/blob/72b5bbba3665f1ae9b2ff016d3d8bf69e7ffc48c/Source/Urho3D/Engine/Engine.cpp#L607

**Reply by laheller** [reply]
> Yes, looks like a bug on my side. I'll look into it, thank you for reporting.

Hi!
@gleblebedev 

Same problem in the latest version of nuget package: 0.3.7.846

BR,

Ladislav

**Reply by gleblebedev** [reply]
> Same problem in the latest version of nuget package: 0.3.7.846
Oh, sorry, I though you found a workaround. I'll look into it again.

**Comment by gleblebedev** [comment]
this should work:
```
            WindowMode windowMode = WindowMode.Windowed;
            switch (windowMode)
            {
                case WindowMode.Windowed:
                    EngineParameters[Urho3D.EpBorderless] = false;
                    EngineParameters[Urho3D.EpWindowResizable] = true;
                    break;
                case WindowMode.Fullscreen:
                    EngineParameters[Urho3D.EpFullScreen] = true;
                    EngineParameters[Urho3D.EpWindowResizable] = false;
                    break;
                case WindowMode.Borderless:
                    EngineParameters[Urho3D.EpBorderless] = true;
                    EngineParameters[Urho3D.EpWindowResizable] = false;
                    break;
            }
```

Tell me if it solves the problem for you. I'll add it to template if it is enough.

**Reply by laheller** [reply]
@gleblebedev 

Now it at least partially works. It starts the windowed mode, but the window itself

- is maximized
- ignores my `EngineParameters[Urho3D.EpWindowWidth] = 1280` and `EngineParameters[Urho3D.EpWindowHeight] = 800` settings
- ignores my `EngineParameters[Urho3D.EpWindowTitle] = "Ladsi"` setting, so the title is empty

**Comment by gleblebedev** [comment]
Window maximized could be fixed by
```
EngineParameters[Urho3D.EpWindowMaximize] = false;
```
The size should work then.

And I'm not sure what happened to window title, it is missing indeed. It should work due to
```
graphics->SetWindowTitle(GetParameter(EP_WINDOW_TITLE).GetString());
```
@eugeneko any ideas?

**Comment by gleblebedev** [comment]
It seems like a bug in the engine.

As a workaround you can set window title in Start() method of Application:
```
            Context.Graphics.WindowTitle = "Bla";
```

**Reply by laheller** [reply]
@gleblebedev @eugeneko 

OK, calling `EngineParameters[Urho3D.EpWindowMaximize] = false;` did the trick. This should be added to the template I think.
Also `Context.Graphics.WindowTitle = "Ladsi";` works as expected.

Thanks for answers!
